src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs

```diff
@@ -140,5 +141,34 @@ public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
                     throw new NotSupportedException("Not supported type " + value.GetType());
             }
         }
+
+        public static void WriteBinaryDataValue(this Utf8JsonWriter writer, BinaryData value)
+        {
+            ReadOnlySpan<char> inputValue = UTF8Encoding.UTF8.GetChars(value.ToArray()).AsSpan();
+            // Check if it is possible to serialize the value as json value.
+            // Possible match kinds: Json Object, Json Array, Json value false, Json value true, Json number.
+            if (inputValue.StartsWith("{".AsSpan(), StringComparison.InvariantCultureIgnoreCase) && (inputValue.EndsWith("}".AsSpan(), StringComparison.InvariantCultureIgnoreCase) || inputValue.EndsWith($"}}{Environment.NewLine}".AsSpan(), StringComparison.InvariantCultureIgnoreCase)) ||
+                inputValue.StartsWith("[".AsSpan(), StringComparison.InvariantCultureIgnoreCase) && (inputValue.EndsWith("]".AsSpan(), StringComparison.InvariantCultureIgnoreCase) || inputValue.EndsWith($"]{Environment.NewLine}".AsSpan(), StringComparison.InvariantCultureIgnoreCase)) ||
+                inputValue.Equals("true".AsSpan(), StringComparison.Ordinal) || inputValue.Equals("false".AsSpan(), StringComparison.Ordinal) ||
+                inputValue.Length > 0 && Char.IsDigit(inputValue[0]) && Char.IsDigit(inputValue[inputValue.Length - 1]))
+            {
+                try
+                {
+#if NET6_0_OR_GREATER
+				    writer.WriteRawValue(value);
+#else
+                    JsonSerializer.Serialize(writer, JsonDocument.Parse(value.ToString()).RootElement);
+#endif
+                }
+                catch (JsonException)
+                {
+                    writer.WriteStringValue(inputValue);
+                }
+            }
+            else
+            {
+                writer.WriteStringValue(inputValue);
```

Comment:
If we are always writing string value do we ever want to be able to serialize `"foo": 51`?

Reply:
Our logic here will check if the first byte and last byte is between `0` and `9`, if it is, we will parse it as json number. There fore if you use an integer to instantiate the BinaryData, no matter you do it by `BinaryData.FromString("51")` or `BinaryData.FromObjectAsJson(51)`, the result will be `"foo": 51`